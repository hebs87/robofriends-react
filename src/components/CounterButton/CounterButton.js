import {Component} from 'react';

class CounterButton extends Component {
  constructor(props) {
    super(props);

    this.state = {
      count: 0,
    };
  };

  shouldComponentUpdate(nextProps, nextState, nextContext) {
    return this.state.count !== nextState.count;
  };

  render() {
    const {count} = this.state;
    const {color} = this.props;

    return (
      <button
        color={color}
        onClick={() => {this.setState(state => ({count: state.count + 1}))}}
      >
        Count: {count}
      </button>
    );
  };
}

export default CounterButton;
