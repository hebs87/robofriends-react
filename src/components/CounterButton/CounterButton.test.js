import {shallow} from 'enzyme';
import CounterButton from "./CounterButton";

describe('CounterButton component test suite', () => {
  it('expect to render CounterButton component', () => {
    const mockColor = 'red';
    // shallow renders just the target component and not any nest components that are rendered within it
    // toMatchSnapshot() takes a snapshot of the component and tests it is rendering as it should - any change to the
    // component will then cause the test to fail, as it will no longer match the snapshot
    expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
  });

  it('correctly increments the counter', () => {
    const mockColor = 'red';
    const wrapper = shallow(<CounterButton color={mockColor} />);
    // Get the element in the component by id, so we can simulate intended action (click in this instance)
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count: 1});
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count: 3});
    // No keypress event in the component so this should not change the state
    wrapper.find('[id="counter"]').simulate('keypress');
    expect(wrapper.state()).toEqual({count: 3});
    // We can also check particular props
    expect(wrapper.props().color).toEqual('red');
  });
});
