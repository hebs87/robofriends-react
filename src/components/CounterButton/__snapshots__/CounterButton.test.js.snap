// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton component test suite expect to render CounterButton component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
